src/factorFinder/FactorComputeEngineImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.example.factorfinder;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FactorComputeEngineImpl implements FactorComputeEngine {
+  String delimiter = ",";
+
+  /**
+   * Input some number from the user
+   * returns array of the factors
+   */
+  public int[] getFactors(int number) {
+    List<Integer> factors = new ArrayList<>();
+    for (int i = 1; i <= number; i++) { // Start from 1 and go to number
+      if (number % i == 0) {
+        factors.add(i); // Add factor to list
+      }
+    }
+    // Convert ArrayList to array
+    //.mapToInt() operation converts each Integer object in the stream to its
+    // corresponding int primitive value. (Integer::intValue) is shortHand for
+    // the intValue() method one EACH Integer Object in the stream. Integer ->
+    // Integer.intValue() is what it should look like when expanded .toArray
+    // collects the elements in the IntStream and returns them as an int[]
+    // array.
```

Comment:
Do you guys think my comment for this line helps you understand it, or is this too much?

Reply:
Yeah it works

Reply:
I think it helps with readability